OpenRA.Mods.Common/Widgets/Logic/Editor/ActorEditLogic.cs

```diff
@@ -340,14 +340,16 @@ ScrollItemWidget SetupItem(PlayerReference option, ScrollItemWidget template)
 							initContainer.Bounds.Height += dropdownContainer.Bounds.Height;
 							dropdownContainer.Get<LabelWidget>("LABEL").GetText = () => ddo.Name;
 
-							var editorActionHandle = new EditorActorOptionActionHandle<string>(ddo.OnChange, ddo.GetValue(actor));
+							var labels = ddo.GetLabels(actor);
```

Comment:
doesn't this defeat the purpose of being dynamic?

Reply:
But the labels are still cached, just the cache is not directly inside `EditorActorDropdown` as a field, but instead it's in a closure in one of the constructors.